core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -97,6 +98,17 @@ enum Consider {
 	RejectFuture,
 	/// Message cannot be evaluated. Reject.
 	RejectOutOfScope,
+	/// Message is redundant and peer should have known. Reject.
+	RejectWillfulRedundant,
+}
+
+/// A tally of votes,
+/// used both to tally incoming votes we receive and outgoing votes we sent out.
+#[derive(Debug, Default)]
+pub(crate) struct VoteTally {
+	pre_commits: usize,
```

Comment:
nitpick, feel free to ignore, but everywhere else we use `prevotes` and `precommits`.